dali/base/dadfs.cpp

```diff
@@ -1396,15 +1398,24 @@ class CDelayedDelete: public CInterface
                 {
                     case SDSExcpt_LockTimeout:
                     case SDSExcpt_LockHeld:
-                        e->Release();
+                        timeoutException.setown(e);
                         break;
                     default:
                         throw;
                 }
             }
             file.clear();
+            unsigned remaining;
+            if (timer.timedout(&remaining))
+            {
+                StringBuffer timeoutText;
+                throwStringExceptionV(-1, "Failed to remove %s: %s", logicalname, timeoutException->errorMessage(timeoutText).str());
+            }
             PROGLOG("CDelayedDelete: pausing due to locked file = %s", logicalname);
-            Sleep(SDS_TRANSACTION_RETRY/2+(getRandom()%SDS_TRANSACTION_RETRY));
+            unsigned sleepTime = SDS_TRANSACTION_RETRY/2+(getRandom()%SDS_TRANSACTION_RETRY);
+            if (sleepTime>remaining)
```

Comment:
nit: Logic is fine but imho would be more clear if there was a branch for when timeoutms is INIFINITE.

Reply:
agree, I will change to handle INFINITE separately